lib/src/main/java/io/ably/lib/transport/Hosts.java

```diff
@@ -60,9 +60,19 @@ public Hosts(String primaryHost, String defaultHost, ClientOptions options) thro
             setPrimaryHost(defaultHost);
         }
         fallbackHostsUseDefault = options.fallbackHostsUseDefault;
-        if (options.fallbackHosts == null) {
-            fallbackHosts = Arrays.copyOf(Defaults.HOST_FALLBACKS, Defaults.HOST_FALLBACKS.length);
-            fallbackHostsIsDefault = true;
+        if (options.tlsPort != 0 || options.port != 0) {
+            fallbackHosts = new String[]{};
```

Comment:
As stated in, https://github.com/ably/ably-ruby/pull/196 we will not use defaultFallbackHosts or EnvFallbackHosts when port and tlsport are set. But, we will set fallbackHosts if passed in options. The current implementation ignores options.fallbackHosts when port and tlsport are set.

Reply:
You might like to take a look at https://github.com/ably/ably-js/pull/689/files for reference

Reply:
I didn't check the latest PR, based on a mutual understanding on old js PR.